homeassistant/components/litterrobot/coordinator.py

```diff
@@ -20,40 +20,42 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-UPDATE_INTERVAL_SECONDS = 60 * 5
+UPDATE_INTERVAL = timedelta(minutes=5)
 
+type LitterRobotConfigEntry = ConfigEntry[LitterRobotDataUpdateCoordinator]
 
-class LitterRobotHub:
-    """A Litter-Robot hub wrapper class."""
 
-    def __init__(self, hass: HomeAssistant, data: Mapping[str, Any]) -> None:
-        """Initialize the Litter-Robot hub."""
-        self._data = data
-        self.account = Account(websession=async_get_clientsession(hass))
+class LitterRobotDataUpdateCoordinator(DataUpdateCoordinator[None]):
+    """The Litter-Robot data update coordinator."""
 
-        async def _async_update_data() -> bool:
-            """Update all device states from the Litter-Robot API."""
-            await self.account.refresh_robots()
-            return True
+    config_entry: LitterRobotConfigEntry
 
-        self.coordinator = DataUpdateCoordinator(
+    def __init__(
+        self, hass: HomeAssistant, config_entry: LitterRobotConfigEntry
+    ) -> None:
+        """Initialize the Litter-Robot data update coordinator."""
+        super().__init__(
             hass,
             _LOGGER,
+            config_entry=config_entry,
             name=DOMAIN,
-            update_method=_async_update_data,
-            update_interval=timedelta(seconds=UPDATE_INTERVAL_SECONDS),
+            update_interval=UPDATE_INTERVAL,
         )
 
-    async def login(
-        self, load_robots: bool = False, subscribe_for_updates: bool = False
-    ) -> None:
-        """Login to Litter-Robot."""
+        self.account = Account(websession=async_get_clientsession(hass))
+
+    async def _async_update_data(self) -> None:
+        """Update all device states from the Litter-Robot API."""
+        await self.account.refresh_robots()
```

Comment:
Can't this raise?

Reply:
The library catches the exceptions right now and just logs an error...